apps/examples/testcase/ta_tc/task_manager/itc/itc_task_manager_main.c

```diff
@@ -0,0 +1,370 @@
+	TC_ASSERT_GEQ("task_manager_register_builtin", tm_sample_handle2, 0);
+
+	int ret;
+	ret = task_manager_start(tm_sample_handle2, TM_RESPONSE_WAIT_INF);
+	TC_ASSERT_EQ_CLEANUP("task_manager_start", ret, OK, unregister_tasks());
+
+	sleep(1);
+
+	ret = task_manager_unregister(tm_sample_handle, TM_RESPONSE_WAIT_INF);
+	TC_ASSERT_EQ("task_manager_unregister", ret, OK);
+
+	ret = task_manager_unregister(tm_sample_handle2, TM_RESPONSE_WAIT_INF);
+	TC_ASSERT_EQ("task_manager_unregister", ret, OK);
+
+	TC_SUCCESS_RESULT();
+}
+
+void itc_task_manager_user1_main(void)
+{
+	int ret;
+
+	ret = task_manager_start(tm_sample_handle, TM_RESPONSE_WAIT_INF);
+	TC_ASSERT_NEQ("task_manager_start", ret, OK);
+
+}
+
+void cleanup_task1(void)
+{
+	unregister_task1 = true;
+}
+
+void unregister_tasks_1(void)
+{
+	int ret;
+
+	ret = task_manager_unregister(tm_sample_handle, TM_RESPONSE_WAIT_INF);
+	TC_ASSERT_EQ("task_manager_unregister", ret, OK);
+
+	int tm_sample_handle2;
+
+	tm_appinfo_list_t *sample_list_info;
+
+	sample_list_info = ((tm_appinfo_list_t *)task_manager_getinfo_with_name("task2", TM_RESPONSE_WAIT_INF));
+	TC_ASSERT_NEQ("task_manager_getinfo_with_name", sample_list_info, NULL);
+
+	tm_sample_handle2 = sample_list_info->task.handle;
+	task_manager_clean_infolist(&sample_list_info);
+	TC_ASSERT_NEQ("task_manager_clean_infolist", (tm_appinfo_list_t *)&sample_list_info, NULL);
+
+	ret = task_manager_unregister(tm_sample_handle2, TM_RESPONSE_WAIT_INF);
+	TC_ASSERT_EQ("task_manager_unregister", ret, OK);
+
+}
+
+void itc_task_manager_pause_task1(void)
+{
+	int ret;
+
+	ret = task_manager_pause(tm_sample_handle, TM_RESPONSE_WAIT_INF);
+	TC_ASSERT_EQ_CLEANUP("task_manager_pause", ret, OK, cleanup_task1());
+
+	ret = task_manager_resume(tm_sample_handle, TM_RESPONSE_WAIT_INF);
+	TC_ASSERT_EQ_CLEANUP("task_manager_resume", ret, OK, cleanup_task1());
+
+	ret = task_manager_restart(tm_sample_handle, TM_RESPONSE_WAIT_INF);
+	TC_ASSERT_EQ_CLEANUP("task_manager_restart", ret, OK, cleanup_task1());
+
+	ret = task_manager_stop(tm_sample_handle, TM_RESPONSE_WAIT_INF);
+	TC_ASSERT_EQ_CLEANUP("task_manager_stop", ret, OK, cleanup_task1());
+
+	ret = task_manager_unregister(tm_sample_handle, TM_RESPONSE_WAIT_INF);
+	TC_ASSERT_EQ_CLEANUP("task_manager_unregister", ret, OK, cleanup_task1());
+
+}
+
+int itc_task_manager_pause_task1_main(int argc, char *argv[])
+{
+	itc_task_manager_pause_task1();
+	return 0;
+}
+
+void itc_task_manager_check_group_permission_p(void)
+{
+	int tm_sample_handle2;
+
+	tm_sample_handle = task_manager_register_task("task1", 100, 1024, not_builtin_task_1, NULL, TM_APP_PERMISSION_GROUP, TM_RESPONSE_WAIT_INF);
+	TC_ASSERT_GEQ("task_manager_register_task", tm_sample_handle, 0);
+
+	tm_sample_handle2 = task_manager_register_task("task2", 100, 1024, itc_task_manager_pause_task1_main, NULL, TM_APP_PERMISSION_DEDICATE, TM_RESPONSE_WAIT_INF);
+	TC_ASSERT_GEQ("task_manager_register_task", tm_sample_handle2, 0);
```

Comment:
is it okay not to unregister task1?

Reply:
If task1 does not unregister in itc_task_manager_pause_task1 then it will unregister in itc_task_manager_check_group_permission_p.